js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AutomaterializeMiddlePanel.tsx

```diff
@@ -1,352 +0,0 @@
-import {useQuery} from '@apollo/client';
-import {Box, Colors, Subheading, Icon, NonIdealState} from '@dagster-io/ui';
-import * as React from 'react';
-import styled from 'styled-components/macro';
-
-import {ErrorWrapper} from '../../app/PythonErrorInfo';
-import {AssetKey} from '../types';
-
-import {AutomaterializeRequestedPartitionsLink} from './AutomaterializeRequestedPartitionsLink';
-import {AutomaterializeRunTag} from './AutomaterializeRunTag';
-import {GET_EVALUATIONS_QUERY} from './GetEvaluationsQuery';
-import {EvaluationOrEmpty} from './types';
-import {
-  AutoMateralizeWithConditionFragment,
-  GetEvaluationsQuery,
-  GetEvaluationsQueryVariables,
-} from './types/GetEvaluationsQuery.types';
-
-const isRequestCondition = (
-  condition: AutoMateralizeWithConditionFragment,
-): condition is AutoMateralizeWithConditionFragment => {
-  switch (condition.__typename) {
-    case 'MissingAutoMaterializeCondition':
-    case 'DownstreamFreshnessAutoMaterializeCondition':
-    case 'FreshnessAutoMaterializeCondition':
-    case 'ParentMaterializedAutoMaterializeCondition':
-      return true;
-    default:
-      return false;
-  }
-};
-
-type ConditionType = AutoMateralizeWithConditionFragment['__typename'];
-
-const extractRequestedPartitionKeys = (conditions: AutoMateralizeWithConditionFragment[]) => {
-  let requested: string[] = [];
-  let skippedOrDiscarded: string[] = [];
-
-  conditions.forEach((condition) => {
-    const didRequest = isRequestCondition(condition);
-    const partitionKeys =
-      condition.partitionKeysOrError?.__typename === 'PartitionKeys'
-        ? condition.partitionKeysOrError.partitionKeys
-        : [];
-    if (didRequest) {
-      requested = requested.concat(partitionKeys);
-    } else {
-      skippedOrDiscarded = skippedOrDiscarded.concat(partitionKeys);
-    }
-  });
-
-  const skippedOrDiscardedSet = new Set(skippedOrDiscarded);
-  return new Set(requested.filter((partitionKey) => !skippedOrDiscardedSet.has(partitionKey)));
-};
-
-interface Props {
-  assetKey: AssetKey;
-  assetHasDefinedPartitions: boolean;
-  selectedEvaluation?: EvaluationOrEmpty;
-  maxMaterializationsPerMinute: number;
-}
-
-export const AutomaterializeMiddlePanel = ({
-  assetKey,
-  assetHasDefinedPartitions,
-  selectedEvaluation,
-  maxMaterializationsPerMinute,
-}: Props) => {
-  const {data, loading, error} = useQuery<GetEvaluationsQuery, GetEvaluationsQueryVariables>(
-    GET_EVALUATIONS_QUERY,
-    {
-      variables: {
-        assetKey,
-        cursor: selectedEvaluation?.evaluationId
-          ? (selectedEvaluation.evaluationId + 1).toString()
-          : undefined,
-        limit: 2,
-      },
-    },
-  );
```

Comment:
It's not clear to me that we actually need this query -- the page already retrieves this data, and the selected evaluation is passed to the middle panel. Perhaps I'm missing something?

Reply:
Originally this was needed because we wanted the ability for you to go to the next page without changing the selected evaluation. In that case, since the page is changed then the data for the selected evaluation won't be present. I believe Johann changed the logic to just pick the first item (so switching pages also switches your selected evaluation which maybe is not a big deal).

Reply:
I think the initial thought for a second query was that we could query the minimum amount for the left panel (just numRequested etc.) and all the conditions info for the main panel. Probably premature optimization

Reply:
@johannkm It's also for linking directly to an evaluation. Right now if the evaluation isn't in the left panel (because it's no longer on the first page) and you give someone a link to the evaluation, then when the evaluation falls outside of the first page (there's no cursor in the URL for the first page), then the link stops working.

Reply:
If we include the cursor even on page 1 then the URL linking would still work, its not super straight forward to do without causing a refetch though.